hagelslag/data/ZarrModelGrid.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import pandas as pd
+from pandas import date_range
+from os.path import exists, join
+import s3fs
+import xarray as xr
+from datetime import timedelta
+
+
+class ZarrModelGrid(object):
+    """
+    Base class for reading 2D model output grids from HRRR Zarr data streamed off of AWS.
+
+    Given an AWS bucket name, loads the values of a single variable from a model run. Supports model output in
+    Zarr format.
+
+    Attributes:
+        path (str): Base Path for AWS Bucket
+        run_date (ISO date string or datetime.datetime object): Date of the initialization time of the model run.
+        start_date (ISO date string or datetime.datetime object): Date of the first timestep extracted.
+        end_date (ISO date string or datetime.datetime object): Date of the last timestep extracted.
+        freqency (str): spacing between model time steps.
+        valid_dates: DatetimeIndex of all model timesteps
+        forecast_hours: array of all hours in the forecast
+        file_objects (list): List of the file objects for each model time step
+    """
+    def __init__(self,
+                 path,
+                 run_date,
+                 start_date,
+                 end_date,
+                 variable,
+                 frequency="1H"):
+        self.path = path
+        self.variable = variable
+        self.run_date = pd.to_datetime(run_date)
+        self.start_date = pd.to_datetime(start_date)
+        self.end_date = pd.to_datetime(end_date)
+        self.frequency = frequency
+        self.valid_dates = date_range(start=self.start_date,
+                                         end=self.end_date,
+                                         freq=self.frequency)
+        print(self.run_date)
+        print(type(self.run_date))
+        self.forecast_hours = (self.valid_dates - self.run_date).astype("timedelta64[h]").astype(int)
+
+
+    def load_data(self):
+
+        if self.run_date.hour not in [0, 6, 12, 18]:
```

Comment:
My only concern with modifying the end_date within load_data is that some users may only want to extract a subset of hours and not the entire model run. I'd rather people account for this issue in the config file by specifying a different end hour for the off hour HRRR runs rather than changing configuration silently silently in load_data. If you are changing things, you should at least issue a warning when doing so.

Reply:
That makes sense. I added the configuration options. Should be good to go.